project/src/components/film-card/film-card.tsx

```diff
@@ -1,23 +1,46 @@
 import { Link } from 'react-router-dom';
+import VideoPlayer from '../video-player/video-player';
+import { useState } from 'react';
 
 type FilmCardProps = {
   id: number,
   previewImage: string,
   name: string,
   isActive: boolean,
-  makeCardActive: (id: number) => void
+  makeCardActive: (id: number) => void,
+  previewVideoLink: string
 };
 
 function FilmCard(props: FilmCardProps): JSX.Element {
-  const { id, previewImage, name, isActive, makeCardActive } = props;
+  const { id, previewImage, name, isActive, makeCardActive, previewVideoLink } = props;
+  const [isShowVideo, setIsShowVideo] = useState(false);
+
+  const onMouseOverHandle = () => {
+    makeCardActive(id);
+    setIsShowVideo(true);
+  };
+
+  const onMouseOutHandle = () => {
+    setIsShowVideo(false);
+  };
+
   return (
-    <article className={`small-film-card catalog__films-card ${ isActive ? 'active' : ''}` } onMouseOver={() => makeCardActive(id)}>
-      <div className="small-film-card__image">
-        <img src={ previewImage } alt={ name } width="280" height="175"/>
-      </div>
-      <h3 className="small-film-card__title">
-        <Link className="small-film-card__link" to={`/films/${id}`}>{ name }</Link>
-      </h3>
+    <article
+      className={`small-film-card catalog__films-card ${ isActive ? 'active' : ''}` }
+      onMouseOver={() => onMouseOverHandle()}
+      onMouseOut={() => onMouseOutHandle()}
+    >
```

Comment:
здесь можно сократить до 
```suggestion
      onMouseOver={onMouseOverHandle}
      onMouseOut={onMouseOutHandle}
    >
```